### ISSUE TITLE
React 18 support for twind/next

### ISSUE THREAD

**Issue by venables** [tw-in-js/use-twind-with#31]
Next.js supports [React 18](https://nextjs.org/docs/advanced-features/react-18/overview) by default now, but `@twind/next` is defined to max out at `react ^17` 

I believe the library is compatible with react 18 as-is, so it should be an easy fix -- by modifying the [react peerDependency in package.json](https://github.com/tw-in-js/use-twind-with/blob/3a846bf584a0def75c8efa025180c3c2cf29ce99/packages/next/package.json#L36) to include ^18, like so:

```diff
-    "react": "^16.6.0 || ^17",
+    "react": "^16.6.0 || ^17 || ^18",
```

**Comment by jmayergit** [comment]
> I believe the library is compatible with react 18 as-is

This library needs to be updated so it's not so easy

https://nextjs.org/docs/advanced-features/custom-document
> To prepare for React 18, we recommend avoiding customizing getInitialProps and renderPage, if possible.
This library customizes getInitialProps and renderPage

https://github.com/vercel/next.js/issues/31225
> Prior to React 18, many CSS-in-JS libraries would just return a script tag for you to insert into your Document at the end of the render...
This library does this

https://github.com/reactwg/react-18/discussions/110
> Library Upgrade Guide: <style> (most CSS-in-JS libs)
This library can be upgraded using the guide provided here

**Comment by jmayergit** [comment]
@sastan can we get your input on this? Thank you

**Comment by swoppy** [comment]
any update on this? `@twind/next` seems not working with next version 13.0.0

**Comment by sastan** [comment]
The version range update is published as @twind/next@1.0.10
